flightcode/video/vid: port send_sprop() pad/caps handling to GStreamer 1.0

GStreamer 1.0 replaces gst_element_get_pad() and GST_PAD_CAPS(). Update the
logic to use gst_element_get_static_pad() and gst_pad_get_current_caps().

--- a/flightcode/video/vid/vidlaunch.cpp
+++ b/flightcode/video/vid/vidlaunch.cpp
@@ -317,6 +317,7 @@
     GstStructure *str;
     GstPad *pad;
     int len;
+    int ret = -1;
 
     // Always send to ARTOO's IP on the OOB port.
     memset((char *)&addr, 0, sizeof(addr));
@@ -326,31 +327,38 @@
 
     /*Attempt to pull of the sprop-parameter-sets from the
      * payloader.  If they're OK, send them along */
-    if ((pad = gst_element_get_pad(payloader, "src")) != NULL) {
-        if ((caps = GST_PAD_CAPS(pad)) != NULL) {
-            if ((str = gst_caps_get_structure(caps, 0)) != NULL) {
+    pad = gst_element_get_static_pad(payloader, "src");
+    if (pad != NULL) {
+        caps = gst_pad_get_current_caps(pad);
+        if (caps != NULL) {
+            str = gst_caps_get_structure(caps, 0);
+            if (str != NULL) {
                 gc = gst_structure_get_string(str, "sprop-parameter-sets");
                 if (gc == NULL) {
                     syslog(LOG_ERR, "Unable to get sprop from gstreamer");
                     len = 0;
-                } else
+                } else {
                     len = strlen(gc);
+                }
 
                 if (len > 0) {
                     if (sendto(fd, (const char *)gc, len, 0, (struct sockaddr *)&addr,
                                sizeof(addr)) < len) {
                         syslog(LOG_ERR, "Unable to send data on OOB port\n");
-                        return -1;
-                    } else
-                        return len;
+                        ret = -1;
+                    } else {
+                        ret = len;
+                    }
                 }
             }
+            gst_caps_unref(caps);
         }
+        gst_object_unref(pad);
     }
 
     /* If we sent -1 bytes its entirely possible that
      * the pipeline did not have a prop to get yet */
-    return -1;
+    return ret;
 }
 
 /* The main routine. Argumenets are only for gstreamer,

